smart_campus/app/views.py

```diff
@@ -342,8 +374,11 @@ def get_all_stations(request):
             'category': str(station.category),
             'location': station.location.get_coords(),
             'image': {
-                'primary': 'http://{0}{1}'.format(request.get_host(), station.primary_image_url),
-                'others': ['http://{0}/{1}'.format(request.get_host(), img.image.url)
+                'primary':
+                    'http://{0}{1}'.format(request.get_host(), StationImage.objects.filter(station=station, is_primary=True).first().image.url)
+                    if StationImage.objects.filter(station=station, is_primary=True).exists()
+                    else '',
+                'others': ['http://{0}{1}'.format(request.get_host(), img.image.url)
                            for img in StationImage.objects.filter(station=station, is_primary=False)]
```

Comment:
`primary` and `other` are far too complicated.
Save them as variables

Reply:
I don't know what you mean, can you explain in another way again?

Reply:
@Lee-W

Reply:
Oh, I misunderstood it.
I think you should write the image part as a method in `Station` class.

Reply:
@rapirent

Reply:
Okay, I will add the Station Class method.

Reply:
But I am curious about why primary will return '' ?
anyway, I will follow this.